src/main/java/com/google/sps/utils/UserUtils.java

```diff
@@ -47,25 +47,43 @@
     * @param  entityPropertyValue  the value of the property that is filtered for
     * @return                      a single entity that has the same value for the property in the parameter 
     */
+
+public static Entity initializeUser(String userId, String name, String sessionID, DatastoreService datastore){
+        Entity userEntity = getEntityFromDatastore("user", "userID", userId, datastore);
```

Comment:
You spelled the key "userID" here and "userId" earlier. That's a pretty devious bug, because it'll either cause user data to be overwritten by accident, or prevent user data from being initialized, depending on which spelling is correct.

This is why I've been recommending that you write a `User` class that lets you not have to deal with `Entity` directly. This is my big "I told you so" moment, and I'm not wasting it. See? I'm gloating.

Reply:
Lmao, I misread and there's no bug. Oh well. My point stands. Mostly